src/StreamrClient.js

```diff
@@ -87,6 +87,8 @@ export default class StreamrClient extends EventEmitter {
             (streamId) => this.getStream(streamId).catch((err) => this.emit('error', err)),
         )
 
+        this.on('error', () => this.disconnect().catch(() => {}))
```

Comment:
Try `this.ensureDisconnected()` instead of suppressing everything in `.catch`? Strong :-1: on ever doing `.catch(() => {})` as described in https://github.com/streamr-dev/streamr-client-javascript/pull/51#issue-266155467